File: app/src/main/java/com/electrosim/GridViewActivity.java
package com.electrosim;

import com.electrosim.adapter.ImageAdapter;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.GridView;
import android.widget.AdapterView.OnItemClickListener;

public class GridViewActivity extends Activity {

	GridView gridView;

	static final String[] MOBILE_OS = new String[] { "circuit1", "circuit2",
			"circuit3", "circuit2" };

	@Override
	public void onCreate(Bundle savedInstanceState) {

		super.onCreate(savedInstanceState);
		setContentView(R.layout.main2);

		gridView = (GridView) findViewById(R.id.gridView1);

		gridView.setAdapter(new ImageAdapter(this, MOBILE_OS));

		gridView.setOnItemClickListener(new OnItemClickListener() {
			public void onItemClick(AdapterView<?> parent, View v,
					int position, long id) {
				Intent c = new Intent(getBaseContext(), CircuitDrawer.class);
				startActivity(c);

			}
		});

	}

}

File: app/src/main/java/com/electrosim/GridViewActivity.kt
package com.electrosim

import com.electrosim.adapter.ImageAdapter
import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.GridView
import com.electrosim.databinding.Main2Binding

class GridViewActivity : Activity() {
    private lateinit var binding: Main2Binding

    companion object {
        private val MOBILE_OS = arrayOf("circuit1", "circuit2", "circuit3", "circuit2")
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        // Initialize ViewBinding
        binding = Main2Binding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.gridView1.adapter = ImageAdapter(this, MOBILE_OS)

        binding.gridView1.onItemClickListener = AdapterView.OnItemClickListener { _, _, _, _ ->
            val intent = Intent(baseContext, CircuitDrawer::class.java)
            startActivity(intent)
        }
    }
} 